kernel-modules/build/build-wrapper.sh

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+# This script handles decompressing the bundles tarballs and setting environment
+# variables needed by the kernel driver compilation script.
+
+KERNEL_SRC_DIR=""
+
+extract_bundle() {
+    local kernel_version="$1"
+
+    export KERNEL_SRC_DIR
+    KERNEL_SRC_DIR="/scratch/kernel-src/${kernel_version}"
+
+    if [[ ! -d "$KERNEL_SRC_DIR" ]]; then
+        rm -rf /scratch/kernel-src/* 2> /dev/null || true
+        mkdir -p "$KERNEL_SRC_DIR"
+        tar -C "$KERNEL_SRC_DIR" -xzf "/bundles/bundle-${kernel_version}.tgz"
+    fi
+
+    [[ -f "${KERNEL_SRC_DIR}/BUNDLE_BUILD_DIR" ]] || {
+        echo "No BUNDLE_BUILD_DIR entry found in kernel source bundle!"
+        return 1
+    }
+
+    [[ -f "${KERNEL_SRC_DIR}/BUNDLE_UNAME" ]] || {
+        echo "No BUNDLE_UNAME entry found in kernel source bundle!"
+        return 1
+    }
+
+    return 0
+}
+
+export_env_variables() {
+    export bundle_uname
+    export bundle_version
+    export bundle_major
+    export bundle_distro
+    export kernel_build_dir
+
+    bundle_uname="$(cat "${KERNEL_SRC_DIR}/BUNDLE_UNAME")"
+    bundle_version="$(cat "${KERNEL_SRC_DIR}/BUNDLE_VERSION")"
+    bundle_major="$(cat "${KERNEL_SRC_DIR}/BUNDLE_MAJOR")"
+    bundle_distro="$(cat "${KERNEL_SRC_DIR}/BUNDLE_DISTRO")"
+    kernel_build_dir="${KERNEL_SRC_DIR}/$(cat "${KERNEL_SRC_DIR}/BUNDLE_BUILD_DIR")"
```

Comment:
I'm not familiar with this kind of bash. Does this basically mean that these variables are exported with the values set in this function? Apologies if silly question 😅

Reply:
It does, yes. After they are defined here they will be picked up by any spawned subshells